src/lib/jss.js

```diff
@@ -78,7 +78,13 @@ JSS.combineSelectors = function (extracted/*:Extracted*/)/*:Extracted*/ {
     for (var block in extracted) {
         extracts = extracted[block];
         for (var i = 0, l = extracts.length; i < l; i += 1) {
+            // If this selector has an escaped colon, we can't safely combine it
+            // with another selector since it will break in IE < 8
+            if (extracts[i].selector && extracts[i].selector.indexOf('\:') > -1) { continue; }
             for(var j = i + 1; j < l; j += 1) {
+                // If this selector has an escaped colon, we can't safely combine it
+                // with another selector since it will break in IE < 8
+                if (extracts[j].selector && extracts[j].selector.indexOf('\:') > -1) { continue; }
```

Comment:
do we really need the nested loop? If we can use a single loop it would increase the performance, also line 83 and 87 are pretty much the same i am wondering if we can remove this duplication.

Problem: `extracted` seems to be an array of objects, the challenge we have is to find the keys that map to the same value in order to create the selector and to do that we take current `O(n^2)`.

Suggestion: we can think about this as an inverted hash where the key is the style we are applying and the value is an array of selectors that match that value every time you find a new selector we just push it to the corresponding style, the object could look something like:

```js
{
   'display: inline-block;': ['.D\(ib\)', '.IbBox', 'a[class*=LineClamp]']
}
```

ps: I don't know if that suggestion would work better so fell free to ignore it.